app/src/main/java/eu/quelltext/mundraub/activities/SettingsActivity.java

```diff
@@ -195,6 +206,50 @@ public void onClick(View v) {
 
     }
 
+    private void buttonStartMapDownloadClicked(){
+        notification.setSmallIcon(R.mipmap.ic_launcher);
+        notification.setTicker("Fruit Radar");
+        notification.setWhen(System.currentTimeMillis());
+        notification.setContentTitle("Offline map updated");
+        notification.setContentText("Download in progress!");
+
+        final int max_progress = 100;
+        int current_progress = 0;
+
+        notification.setProgress(max_progress, current_progress, false );
+
+        Intent intent= new Intent(this, SettingsActivity.class);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this,0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+        notification.setContentIntent(pendingIntent);
+
+        NotificationManager notificationManager= (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+        notificationManager.notify(uniqueID, notification.build());
+
+        Thread thread = new Thread(){
+            @Override
+            public void run() {
+                int count=0;
+                try {
+                    while (count <= 100) {
```

Comment:
Thanks for showing this code. This is probably a good prototype for how the progress can be implemented. #147 
For this to go live, I would expect it to display the actual progress made. 
I think, you can put code into ProgressUpdate https://github.com/niccokunzmann/mundraub-android/blob/b3c97c078d8eefdc34653f00eea5601520194473/app/src/main/java/eu/quelltext/mundraub/activities/SettingsActivity.java#L65 
Putting it there allows updating the progress bar in a uniform way.